hs/app/reach/Main.hs

```diff
@@ -2465,17 +2465,38 @@ instance FromJSON GitHubGistResponse where
   parseJSON = withObject "GitHubGistResponse" $ \o -> GitHubGistResponse <$> o .: "html_url"
 
 support :: Subcommand
-support = command "support" $ info (pure g) d
+support = command "support" $ info h d
   where
-    d = progDesc "Create GitHub gist of index.rsh and index.mjs"
+    d = progDesc "Create a GitHub gist of index.rsh and index.mjs (default), or specify your own files to upload!"
+    h = go <$> supportFromCommandLineHs
+    go SupportToolArgs {sta_so = SupportOpts {}} = do
+      rawArgs <- liftIO getArgs
+      let rawArgs' = dropWhile (/= "support") rawArgs
+      let actualArgs = tailMay rawArgs'
+      case actualArgs of
+        Nothing -> impossible $ "support args do not start with 'support': " <> show rawArgs
+        Just [] -> g -- no arguments? default behavior!
+        Just xs -> validate xs
+    validate :: [FilePath] -> App
+    validate actualArguments = liftIO $ do
+      arrayOfPairs <- mapM z actualArguments
+      let pairs = zip arrayOfPairs actualArguments
+      let invalidPaths = map snd $ filter (null . fst) pairs
+      case invalidPaths of
+        [] -> upload arrayOfPairs
+        _ -> do
+          putStrLn "Couldn't find the following files:\n"
+          mapM_ putStrLn invalidPaths
+          putStrLn "\nNothing uploaded"
     f i c = i .= object
       [ "content" .= if T.null (T.strip $ pack c) then "// (Empty source file)" else c
       , "language" .= ("JavaScript" :: String)
       , "type" .= ("application/javascript" :: String)
       ]
     z i = doesFileExist i >>= \case
       False -> pure []
```

Comment:
My taste is to remove `validate` and put the error right here

Reply:
[c801f5d](https://github.com/reach-sh/reach-lang/pull/1063/commits/c801f5d28e0c3970649f012929850b2f87399ce1)